### ISSUE TITLE
[Native] Getting build issue after upgrading Presto to latest - 0.288-SNAPSHOT

### ISSUE THREAD

**Issue by agrawalreetika** [prestodb/presto#22649]
I am encountering below error while building `presto-native-execution` after upgrading Presto to latest - 0.288-SNAPSHOT -

> cd /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/_build/debug/presto_cpp/main/thrift && mkdir -p /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/_build/debug/presto_cpp/main/thrift/presto_cpp/main/thrift && /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/dependencies_install/bin/thrift1 --gen mstch_cpp2:include_prefix=.. -o /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/_build/debug/presto_cpp/main/thrift/presto_cpp/main/thrift /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/presto_cpp/main/thrift/presto_thrift.thrift
> dyld[62257]: Symbol not found: __ZN5boost10filesystem6detail9canonicalERKNS0_4pathES4_PNS_6system10error_codeE
>   Referenced from: <03F41F0E-D8E8-3C82-949B-F0B6DB130959> /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/dependencies_install/bin/thrift1
>   Expected in:     <B1D1C4ED-C73D-388C-ACB5-5A10ABEDF634> /opt/homebrew/Cellar/boost/1.85.0/lib/libboost_filesystem-mt.dylib
> /bin/sh: line 1: 62257 Abort trap: 6           /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/dependencies_install/bin/thrift1 --gen mstch_cpp2:include_prefix=.. -o /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/_build/debug/presto_cpp/main/thrift/presto_cpp/main/thrift /Users/reetikaagrawal/Documents/Projects/Oss/prestodb-hivepoc/prestodb/presto-native-execution/presto_cpp/main/thrift/presto_thrift.thrift
> [540/1217] Performing build step for 'libstemmer'

**Comment by czentgr** [comment]
When you "upgraded" - did you rebuild all the dependencies aka ran the `presto-native-execution/scripts/setup-macos.sh` script and rebuild and installed all the deps anew?
The version of the dependencies recently changed to a much later version.

**Comment by agrawalreetika** [comment]
Update -

Yes @czentgr , I was able to get past the issue and successfully build presto-native-execution after doing below -
1. Removed existing _build folder
2. Removed old dependencies which was installed (This is optional, since you still get an option to reinstall with script)
3. Ran the `presto-native-execution/scripts/setup-macos.sh` script  again to install dependencies
4. Ran the build again and it went fine
